posthog/models/async_deletion/delete_events.py

```diff
@@ -36,19 +46,33 @@ def process(self, deletions: list[AsyncDeletion]):
         )
 
         conditions, args = self._conditions(deletions)
-        sync_execute(
-            f"""
-            ALTER TABLE sharded_events
-            ON CLUSTER '{CLICKHOUSE_CLUSTER}'
-            DELETE WHERE {" OR ".join(conditions)}
-            """,
-            args,
-            settings={"max_query_size": MAX_QUERY_SIZE},
-        )
+
+        query_predicate = []
+        for condition in conditions:
+            query_predicate.append(condition)
+
+            # Get estimated  byte size of the query
+            str_predicate = " OR ".join(query_predicate)
+            query_size = len(str_predicate.encode("utf-8"))
+
+            if query_size > MAX_PREDICATE_SIZE:
```

Comment:
What's this for? query needs to be larger than a threshold?

Reply:
oh queries don't have to be bigger than this, this is just a circuit breaker for if it is larger than this it is approaching our query statement size limit and we should split up the delete statements (which it does in that statement)